### DISCUSSION TITLE
Odd performance (low) in Osu!Lazer (mac)

### DISCUSSION THREAD

**Discussion by the-zphr** [ppy/osu#13506] [Q&A] [answered]
**Describe the bug:** 
Very strange and low performance in lobby, beatmap selection (the lowest there) and while playing beatmaps. so basically everything, also i discovered healthbar bug
**Screenshots or videos showing encountered issue:** 
Google Photos: https://photos.app.goo.gl/eU6jF2Xe287kZByj8
**osu!lazer version:** 
2021.612.0
**Logs:**
[database.log](https://github.com/ppy/osu/files/6657450/database.log)
[network.log](https://github.com/ppy/osu/files/6657451/network.log)
[performance.log](https://github.com/ppy/osu/files/6657452/performance.log)
[runtime.log](https://github.com/ppy/osu/files/6657453/runtime.log)

**Comment by bdach** [comment]
* For the skin problem, please attach the skin.
* When it comes to the performance issues in song select the draw thread seems to be most under duress (lot of time spent in `SwapBuffer`). Seeing that you don't have vsync on then it seems that there is indeed room for improvement. An old comment from a while ago [listed that this can be an issue](https://github.com/ppy/osu/issues/3252#issuecomment-416407055) but there have been several improvements since then.

  If possible, please list your specs as they may give us more to work with.

**Comment by the-zphr** [comment, marked as answer]
i dont think its a skin problem, since i tried using other skins.

and my specs:
Macbook Pro 2019 (16inch)
Intel Core i7 (6 coresm 2,6GHz)
16GB Ram DDR4 2667 MHz
AMD Radeon Pro 5300M 4gb vram

**Reply by bdach** [reply]
I have checked several other skins and they were aligned correctly. Please do attach it. Or another one that seems broken.

**Reply by the-zphr** [reply]
Minimalist 2.0 by Krizto
Google Drive https://drive.google.com/drive/folders/1dTEq0NwzC9ytQPaKoyLFynAyiK3c2GBA

normal version, not lite

**Reply by the-zphr** [reply]
also few seconds ago i tested basic skins, same low fps.
so its not a skin problem

**Reply by bdach** [reply]
When I asked for the skin, I was in particular referring to what you called a "healthbar bug" (without specifying what it was). I have just downloaded and imported the skin you linked and the healthbar itself looked fine.

![osu_2021-06-15_20-03-24](https://user-images.githubusercontent.com/20418176/122101585-c4b81c00-ce14-11eb-9c59-2119873f414c.jpg)

It _is_ true that the healthbar background is no longer aligned with the score/accuracy text, but that was never officially supported, skin authors were just relying on elements to be positioned in a particular way. This will not be supported going forward. If you wish to fix the issue, then you can realign the skin background elements yourself.

**Reply by the-zphr** [reply]
alright what about performance?

**Reply by bdach** [reply]
When it comes to the performance issues I'll yield to mac users here to diagnose further. I can say for sure that I am not seeing framerates nearly as low on a 5-year-old notebook, I am seeing mostly solid 120fps in song select (although the computer I'm testing on was considered a higher-range machine back then), so it seems a bit suspicious. It may have something to do with the fact that the game uses opengl but it's different to say for sure without profiling on the actual machine in question.

**Reply by the-zphr** [reply]
everything was fine before update, still tho its first time when i have that low performance

**Reply by the-zphr** [reply]
it seems like... i solved the issue by my own somehow?

**Reply by the-zphr** [reply]
yeah, it works fine now. i solved issue by my own. thank you for trying to help me, my mac needed only to be restarted

**Reply by bdach** [reply]
It's difficult for me to theorise, but you may have experienced thermal throttling of some sort. I'll mark this discussion as resolved but please do report back again if it happens again without any discernible cause.
